### ISSUE TITLE
Question about operation of "--write-auto-sub"

### ISSUE THREAD

**Issue by section83** [ytdl-org/youtube-dl#27306]
## Checklist

- [X] I'm asking a question
- [X] I've looked through the README and FAQ for similar questions
- [X] I've searched the bugtracker for similar questions including closed ones

## Question

What should result from the following examples ?

> youtube-dl -v --write-auto-sub --write-sub "https://www.youtube.com/watch?v=F5Y7BniaRXg"
> youtube-dl -v --write-auto-sub --write-sub --all-subs "https://www.youtube.com/watch?v=F5Y7BniaRXg"

I get the following from both forms:

> youtube-dl -v --write-auto-sub --write-sub --all-subs "https://www.youtube.com/watch?v=F5Y7BniaRXg"
> [debug] System config: []
> [debug] User config: []
> [debug] Custom config: []
> [debug] Command-line args: [u'-v', u'--write-auto-sub', u'--write-sub', u'--all-subs', u'https://www.youtube.com/watch?v=F5Y7BniaRXg']
> [debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
> [debug] youtube-dl version 2020.12.05
> [debug] Python version 2.7.16 (CPython) - Darwin-19.6.0-x86_64-i386-64bit
> [debug] exe versions: ffmpeg 4.3.1-tessus, ffprobe 4.3.1-tessus
> [debug] Proxy map: {}
> [youtube] F5Y7BniaRXg: Downloading webpage
> WARNING: video doesn't have subtitles
> [youtube] F5Y7BniaRXg: Looking for automatic captions
> WARNING: Couldn't find automatic captions for F5Y7BniaRXg
> 

That result seems correct as using --list-subs also returns a warning that there are no subtitles:

> youtube-dl -v --list-subs "https://www.youtube.com/watch?v=F5Y7BniaRXg"      
> [debug] System config: []
> [debug] User config: []
> [debug] Custom config: []
> [debug] Command-line args: [u'-v', u'--list-subs', u'https://www.youtube.com/watch?v=F5Y7BniaRXg']
> [debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
> [debug] youtube-dl version 2020.12.05
> [debug] Python version 2.7.16 (CPython) - Darwin-19.6.0-x86_64-i386-64bit
> [debug] exe versions: ffmpeg 4.3.1-tessus, ffprobe 4.3.1-tessus
> [debug] Proxy map: {}
> [youtube] F5Y7BniaRXg: Downloading webpage
> WARNING: video doesn't have subtitles
> [youtube] F5Y7BniaRXg: Looking for automatic captions
> WARNING: Couldn't find automatic captions for F5Y7BniaRXg

However, when this form is used, auto generated subtitles ARE downloaded:

> youtube-dl -v --write-auto-sub "https://www.youtube.com/watch?v=F5Y7BniaRXg" 
> [debug] System config: []
> [debug] User config: []
> [debug] Custom config: []
> [debug] Command-line args: [u'-v', u'--write-auto-sub', u'https://www.youtube.com/watch?v=F5Y7BniaRXg']
> [debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
> [debug] youtube-dl version 2020.12.02
> [debug] Python version 2.7.16 (CPython) - Darwin-19.6.0-x86_64-i386-64bit
> [debug] exe versions: ffmpeg 4.3.1-tessus, ffprobe 4.3.1-tessus
> [debug] Proxy map: {}
> [youtube] F5Y7BniaRXg: Downloading webpage
> [youtube] F5Y7BniaRXg: Looking for automatic captions
> [debug] Default format spec: bestvideo+bestaudio/best
> [info] Writing video subtitles to: How to recover data from a hard drive (stuck heads - buzzing, clicking, etc)-F5Y7BniaRXg.en.vtt

Other than testing for a warning, is there a way of finding auto generated subtitles ?

Why does combining _--write-auto-sub_ and _--write-sub_ return the warning that there are no auto generated subtitles when there clearly are ?

Should _--list-subs_ return info on auto generated subtitles ?

**Comment by InspiringSand** [comment]
An observation:
Your first example outputs:
[debug] youtube-dl version 2020.12.05
Your third example outputs:
[debug] youtube-dl version 2020.12.02

When I try this on my machine, it doesn't work for either example using version 2020.12.05.

**Comment by section83** [comment]
Yes, I missed that.  Auto-generated captions don't work for that video in 2020.12.09 either.  Auto-generated captions for that video are visible in my web browser so, it looks like a YTDL bug.

**Comment by pipizhang** [comment]
I just tested 2020.12.12 version, auto-generated captions still **doesn't work**. 

> youtube-dl --list-subs "https://www.youtube.com/watch?v=F5Y7BniaRXg" -vv
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--list-subs', 'https://www.youtube.com/watch?v=F5Y7BniaRXg', '-vv']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] youtube-dl version 2020.12.12
[debug] Python version 3.8.5 (CPython) - Linux-5.4.0-54-generic-x86_64-with-glibc2.29
[debug] exe versions: ffmpeg 4.2.4, ffprobe 4.2.4, rtmpdump 2.4
[debug] Proxy map: {}
[youtube] F5Y7BniaRXg: Downloading webpage
**WARNING: video doesn't have subtitles**
[youtube] F5Y7BniaRXg: Looking for automatic captions
**WARNING: Couldn't find automatic captions for F5Y7BniaRXg**
F5Y7BniaRXg has no automatic captions
F5Y7BniaRXg has no subtitles
